### ISSUE TITLE
Preferences do not save

### ISSUE THREAD

**Issue by ziekke** [xournalpp/xournalpp#168]
When I change the preferences and close xournalpp, upon reopening they are back to default.

I tried running xournalpp from command line and when I click "ok" in the preferences dialog, the following shows up:

`error : string is not in UTF-8`

I'm using dev branch

edit:

The metadata root was created (~/.xournalpp) including the settings.xml, but I'm supposing something is going on in the app that's preventing it from writing the file properly when the OK button is clicked.

edit2:

I manually edited the file and made a change, once I opened xournalpp it reverted the change back to default.

**Comment by ziekke** [comment]
Without doing too much.. maybe just opening/reopening the program not it is saving!

I swear it wasn't working and I don't believe I changed anything significant.

It is worth noting that the first changes that "held" were when I enabled "Add Vertical Space" and "Add Horizontal Space" and clicked ok. I noted that the error above didn't show up. From then adding more options appeared to have worked.

I did get the app to crash after inserting a largeish LaTeX statement and resizing it.. but I'll have to post that if I get it again.

**Comment by JoeKays** [comment]
This seems to still not work on Debian stretch (testing) fresh install with poppler source compile and boost on default branch.
On first startup saving the settings results in:
`I/O error: encoder error`
`error: string is not UTF-8`
`encoding error: output conversion failed due to conv error, bytes 0x8D 0x60 0x64 0x55`
0x8D is indeed no UTF-8 character, possibly 0xC3 0x8D would have been the real character.
This results in empy settings.xml file, giving a new error on second startup.

SOLUTION: Use a settings.xml file from someone else, e.g. from 3rd comment: https://github.com/xournalpp/xournalpp/issues/187
After using an already complete file, the encoder error disappears, changing settings works as expected.

**Comment by xiamaz** [comment]
Can confirm this issue on the latest master. Getting a similar error message.
Strangely when I tried strace the save was successful.
```
error : string is not in UTF-8
encoding error : output conversion failed due to conv error, bytes 0xCA 0x4B 0x02 0x2C
I/O error : encoder error
```

**Comment by fkjogu** [comment]
I also still have this problem on 3a86f9f86b623c1e3bd53ecd2d8f2ffa27a3ce72

**Comment by Hoeze** [comment]
Same problem here.

**Comment by kelmes** [comment]
Could you try the current development branch?

**Comment by Hoeze** [comment]
I did not try the "development" branch, but I changed `ISO-8859-1` to `UTF-8` in "src/control/settings/Settings.cpp" and now settings are saving... 
However, this was just trial-and-error, so I am not sure if this fixes the root of that problem.

**Comment by peijunz** [comment]
It seems that if I delete lines from [250](https://github.com/xournalpp/xournalpp/blob/bb3c205d67ce1d79ec3fa36f0373410eea9d7bdc/src/gui/dialog/SettingsDialog.cpp#L250) to [287](https://github.com/xournalpp/xournalpp/blob/bb3c205d67ce1d79ec3fa36f0373410eea9d7bdc/src/gui/dialog/SettingsDialog.cpp#L287) in SettingsDialog.cpp, everything is saved! From my observation (by plugging in couts to track the newHide string), there are some wrong(non-UTF8) string generated between 250-287. 

It seems that these code lines are also not necessary because the new preference does not need the old one?

**Comment by peijunz** [comment]
Today I found that if xournalpp-master is not built with **Gettext**, this bug appears. But with gettext installed it vanishes. 

This is tricky because the error of Gettext is **not fatal** so that `cmake ..` seems to work well at a glance.
```
--   date_time
--   atomic
-- Could NOT find Gettext (missing: XGETTEXT_EXECUTABLE GETTEXT_MSGMERGE_EXECUTABLE GETTEXT_MSGCAT_EXECUTABLE) 
ERRORGettext not found!
Poppler sources exist

Configuration:
	Compiler:                   /usr/bin/c++
	Overlay Scrollbar enabled:  OFF
	Mathtex enabled:            ON
	CppUnit enabled:            OFF

Unstable features:
	Layers sidebar:             OFF

Static libraries:
	Poppler:                    OFF
	Boost:                      OFF

-- Configuring done
-- Generating done
-- Build files have been written to: /home/zpj/Downloads/xournalpp-master/build
``` 

But if your are more careful, you will notice the line
```
 -- Could NOT find Gettext (missing: XGETTEXT_EXECUTABLE GETTEXT_MSGMERGE_EXECUTABLE GETTEXT_MSGCAT_EXECUTABLE) 
ERRORGettext not found!
```

Hope my experience will help others to solve this issue.

**Comment by siliconninja** [comment]
Can confirm that this workaround **temporarily** works (while in the application) until restarting the application.

I also had to delete the .xml file in .xournalpp otherwise I would get an error upon starting the application that:
```
Input is not proper UTF-8, indicate encoding !
Bytes: 0xF0 0xE8 0xB4 0x80 <property name="presentationHideElements" value="�贀�U"/>
** (xournalpp:2917): WARNING **: Settings::load:: doc == null, could not load Settings
```
and the application won't load at all. I think the reason for this is that it's not saving in UTF8 format until the preferences document gets created.

Also, more clarification on the other UTF8 string not found error: When I click on the "edit" menu, this error pops up. It might possibly be due to something happening where a string in the edit menu isn't being read correctly.

For reference, I am using xournalpp-git on the AUR, from the latest push to the master branch, on linux 4.14.27. I tried applying the workaround by building this myself.

You could edit the file manually for now, but right now I can't find a proper fix for the GUI.

**Comment by siliconninja** [comment]
I have tried combining both fixes into my own repo (@Hoeze's `UTF-8` fix and the official pull request to the development branch https://github.com/xournalpp/xournalpp/pull/223) and they seem to work well! Just to be sure, I've deleted the existing settings.xml file to avoid other issues. I also tried it with my existing `ISO-8859-1` encoded file, and it seems to work just fine with the new commits, maybe because the string is being read properly.

I've tried a few different settings (e.g. Xinput, sidebar on the right hand side, and zoom ruler DPI setting) and tested it in the master branch with mathtex flag in Cmake, to be sure all dependencies work. Now the error of "UTF8 string not found" when I click on the edit menu has also disappeared. The preferences are being written properly to settings.xml file in the .xournalpp folder in my home directory.

I've opened a pull request to the master branch for further discussion on this issue and if there is anything else that needs consideration before merging it into master.
